### PR TITLE
Add pyhpc-benchmark isoneutral_mixing

### PR DESCRIPTION
**Pull request by bertmaher**
Some perf results on one core (in ms):

device | eager | jit | jit+fuser
-|-|-|-
cuda | 238 | 176 | --
cpu | 3652 | 3341 | 3046

### CODE REVIEW

File: torchbenchmark/models/pyhpc_isoneutral_mixing/isoneutral_pytorch.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    if device == 'gpu':
+        torch.cuda.synchronize()
+    return out
+
+
+def run(*inputs, device='cpu'):
+    with torch.no_grad():
+        if device == 'gpu':
+            outputs = isoneutral_diffusion_pre_gpu(*inputs)
+            torch.cuda.synchronize()
```
Comment (wconstab) [inline torchbenchmark/models/pyhpc_isoneutral_mixing/isoneutral_pytorch.py:210]: hmm this is good but we also do it again in the benchmark wrapper so i'm not sure if it adds extra overhead we care about.

**Review by wconstab — APPROVED** [review]
awesome! props for finding a cool and different use case.

from my quick skimming, this is basically a numerical simulator for ocean mixing, and so 'training' is not even a relevant concept- i assume that's the reason for you not implementing it?

**Review by xuzhao9 — APPROVED** [review]
LGTM

File: torchbenchmark/models/pyhpc_isoneutral_mixing/isoneutral_pytorch.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+    K_33[2:-2, 2:-2, :-1] = sumx / (4 * dxt[2:-2, None, None]) + \
+        sumy / (4 * dyt[None, 2:-2, None]
+                * cost[None, 2:-2, None])
+    K_33[2:-2, 2:-2, -1] = 0.
+
+    return K_11, K_22, K_33, Ai_ez, Ai_nz, Ai_bx, Ai_by
+
+
+def prepare_inputs(*inputs, device):
```
Comment (Krovatkin) [inline torchbenchmark/models/pyhpc_isoneutral_mixing/isoneutral_pytorch.py:199]: is this used anywhere?

**Review by Krovatkin — APPROVED** [review]
:shipit:

**Comment by bertmaher** [comment]
> awesome! props for finding a cool and different use case.
> 
> from my quick skimming, this is basically a numerical simulator for ocean mixing, and so 'training' is not even a relevant concept- i assume that's the reason for you not implementing it?

Yep, the benchmark suite I'm pulling this from is just using PT for numerical calculations, there's no training stage.

File: torchbenchmark/models/pyhpc_isoneutral_mixing/isoneutral_pytorch.py (hunk 2 above)
Comment (bertmaher) [reply torchbenchmark/models/pyhpc_isoneutral_mixing/isoneutral_pytorch.py:199]: No, I copied this file verbatim from the pyhpc benchmark harness.  I can delete this for clarity.

File: torchbenchmark/models/pyhpc_isoneutral_mixing/isoneutral_pytorch.py (hunk 1 above)
Comment (bertmaher) [reply torchbenchmark/models/pyhpc_isoneutral_mixing/isoneutral_pytorch.py:210]: I don't think this path is actually run, so I'll delete it (and `prepare_inputs` as above)

**Pull request merged**